relay-server/src/actors/store.rs

```diff
@@ -199,51 +199,137 @@ impl StoreService {
             }
         }
 
+        let remote_addr = envelope.meta().client_addr().map(|addr| addr.to_string());
+
+        let kafkamessages = self.extract_kafka_messages(
+            event_item,
+            event_id,
+            scoping,
+            start_time,
+            remote_addr,
+            &mut attachments,
+        );
+
+        for message in kafkamessages {
+            if let Ok(()) = self.produce(topic, scoping.organization_id, &message) {
+                // if message was succesfully sent, make the right log message
+                match message {
+                    KafkaMessage::Attachment(_) => {
+                        metric!(
+                            counter(RelayCounters::ProcessingMessageProduced) += 1,
+                            event_type = "attachment"
+                        );
+                    }
+                    KafkaMessage::Event(_) => metric!(
+                        counter(RelayCounters::ProcessingMessageProduced) += 1,
+                        // the unwrap is safe because in self.extract_kafka_messages() it only
+                        // pushes an event message if the event_item is_some()..
+                        event_type = &(event_item.expect("oh fuck").ty().to_string())
```

Comment:
This is the guilty unwrap I want to remove. If there's no good way I would just use the if let Some() pattern.